src/Microsoft.AspNet.Mvc.Core/ApplicationModels/ActionModel.cs

```diff
@@ -74,6 +78,6 @@ public ActionModel([NotNull] ActionModel other)
 
         public List<ParameterModel> Parameters { get; private set; }
 
-        public AttributeRouteModel AttributeRouteModel { get; set; }
+        public List<IRouteConstraintProvider> RouteConstraints { get; private set; }
```

Comment:
Can we just fix it now in these two places? :smile: (And have a bug for other, as-yet-unknown cases?)